### ISSUE TITLE
Drastically FPS drop on ubuntu 16.04 gnome

### ISSUE THREAD

**Issue by DanialV** [ValveSoftware/Dota-2#1352]
I had ubuntu 16.04 unity and I played dota 2 very smooth.I change my ubuntu 16.04 from unity to gnome and now when I am playing dota I have very serious fps drop from 60 to 3 or 4 fps time to time in game and it makes it unplayable I have Lenovo thinkpad x230t.
lspci:
```
00:00.0 Host bridge: Intel Corporation 3rd Gen Core processor DRAM Controller (rev 09)
00:02.0 VGA compatible controller: Intel Corporation 3rd Gen Core processor Graphics Controller (rev 09)
00:14.0 USB controller: Intel Corporation 7 Series/C210 Series Chipset Family USB xHCI Host Controller (rev 04)
00:16.0 Communication controller: Intel Corporation 7 Series/C210 Series Chipset Family MEI Controller #1 (rev 04)
00:16.3 Serial controller: Intel Corporation 7 Series/C210 Series Chipset Family KT Controller (rev 04)
00:19.0 Ethernet controller: Intel Corporation 82579LM Gigabit Network Connection (rev 04)
00:1a.0 USB controller: Intel Corporation 7 Series/C210 Series Chipset Family USB Enhanced Host Controller #2 (rev 04)
00:1b.0 Audio device: Intel Corporation 7 Series/C210 Series Chipset Family High Definition Audio Controller (rev 04)
00:1c.0 PCI bridge: Intel Corporation 7 Series/C210 Series Chipset Family PCI Express Root Port 1 (rev c4)
00:1c.1 PCI bridge: Intel Corporation 7 Series/C210 Series Chipset Family PCI Express Root Port 2 (rev c4)
00:1c.2 PCI bridge: Intel Corporation 7 Series/C210 Series Chipset Family PCI Express Root Port 3 (rev c4)
00:1d.0 USB controller: Intel Corporation 7 Series/C210 Series Chipset Family USB Enhanced Host Controller #1 (rev 04)
00:1f.0 ISA bridge: Intel Corporation QM77 Express Chipset LPC Controller (rev 04)
00:1f.2 SATA controller: Intel Corporation 7 Series Chipset Family 6-port SATA Controller [AHCI mode] (rev 04)
00:1f.3 SMBus: Intel Corporation 7 Series/C210 Series Chipset Family SMBus Controller (rev 04)
02:00.0 System peripheral: Ricoh Co Ltd MMC/SD Host Controller (rev 07)
03:00.0 Network controller: Intel Corporation Centrino Advanced-N 6205 [Taylor Peak] (rev 34)

```
When I alt tab and switch window to my desktop for 3 or 4 sec and then switch to game fps back to normal  60 fps I really don't think this my graphic problem.

**Comment by gdrewb-valve** [comment]
The only change was switching from Unity to Gnome?  You'll want to ask on forums related to Gnome also in case this is a known issue.

**Comment by DanialV** [comment]
Actually I reinstall ubuntu with gnome desktop and yes this is the only change I had and I saw this [issue](https://github.com/ValveSoftware/Dota-2/issues/1070) related to gnome but the issue was closed due to the bumblebee problem and I don't have NVIDIA graphic card

**Comment by gdrewb-valve** [comment]
If you did a full system reinstall are you sure your graphics driver setup didn't change?  Also when I suggested checking in a Gnome forum I meant something like the forums for the Gnome project.

**Comment by DanialV** [comment]
I setup my graphic card correctly because I also play other games and they run very well and this is the main reason I thought it maybe related to the dota 2

**Comment by gdrewb-valve** [comment]
If Dota didn't change but your system did it seems like the issue is more likely related to the system change?

**Comment by DanialV** [comment]
if you mean here is wrong place for asking my question yes I think so but I am looking for solution and hope someone with same experience can help me to solve it

**Comment by ghost** [comment]
wild guess.

using wayland?

tried xorg?

**Comment by DanialV** [comment]
I tried both xorg and wayland but nothing changed and still have fps drop and it's not all the time but only specific times and again return back to normal

**Comment by r3pek** [comment]
I would say that #1355 and this one are the same thing.

**Comment by gdrewb-valve** [comment]
This predates the badges and shows up in a different way so I think this is probably a different issue.  There's always a chance, though.

**Comment by inzru** [comment]
Hi there, I am having a similar issue. Am on Ubuntu 16.04 with Intel NUC Skull canyon, FPS dropping to basically zero for a period of 10-20 seconds, about 5 times a game. (though FPS in-game counter says 10-15 FPS, the screen is a literal slide show, one frame per two seconds or so.) happens while spectating and when in the menu too. this started happening after the dota 2 update earlier this week. is pretty game ruining.

**Comment by ghost** [comment]
I have this issue in Watch menue, screen freezez to load medals and it happends every 15-20 minutes in game.

**Comment by gdrewb-valve** [comment]
The Dec. 5 update has fixed the Watch tab issue. @DanialV I'm not sure it'll help your case but do you see any improvement?

**Comment by DanialV** [comment]
no, I still have fps drop during a game for about 3 up to 20 sec and then fps back to normal and this happened multiple time during a game

**Comment by DanialV** [comment]
I have found the cause of fps drop during the game I monitored my CPU usage in some time suddenly my all 4 of my CPU cores  goes to 100% 
there are a lot's of dota 2 thread which eats all of CPU cores  `engine experimental drop frame ticket 1 full screen`  hope this information helps to solve this issue

**Comment by kisak-valve** [comment]
Hello @DanialV, with the symptoms described, you should also be keeping an eye on the system temperatures as sudden bursts to 100% cpu usage while under moderately heavy load can also be caused by the cpu being throttled to protect the system from overheating.

**Comment by tukkek** [comment]
Does the FPS drop happen only during big team fights (especially early on in a match) or when one of the heroes uses a spell or item? Because if that's the case I'm pretty sure I've seen this reported already in this repository. Or are the FPS drops completely random?
